solutions/USACO dec15-feast-gold.mdx

```diff
@@ -0,0 +1,52 @@
+---
+id: usaco-574
+source: USACO Gold 2015 December
+title: Fruit Feast
+author: Danh Ta Chi Thanh
+---
+
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_feast_gold_dec15.html)
+
+
+<LanguageSection>
+
+<CPPSection>
+
+## C++ Implementation
+
+```cpp
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	freopen("feast.in","r",stdin);
+	freopen("feast.out","w",stdout);
+	int T,A,B;
+	cin >> T >> A >> B;
+	bool dp[2][T+1];

```

Comment:
`bool dp[2][T+1]{};` zero-initializes as mentioned here https://usaco.guide/bronze/intro-ds?lang=cpp#arrays

Reply:
Could I move the dp array into global variable (it will be like bool dp[2][5000001];

Reply:
ok